audit-checklists/Wormhole.md

```diff
@@ -0,0 +1,672 @@
+
+1. Message Passing 
+2. Token Bridge
+3. Native Token Transfers
+4. CCTP
+5. Relayers
+
+### Direct `publishMessage` Usage
+
+Protocols can publish messages directly through the Wormhole-Core contract's `publishMessage` function:
+
+```solidity
+function publishMessage(
+    uint32 nonce,
+    bytes memory payload,
+    uint8 consistencyLevel
+) external payable returns (uint64 sequence);
+```
+
+Key parameters and considerations:
+
+1. `nonce`:
+   - Must be unique for each bridge message and caller
+   - Protocols must implement proper nonce generation
+   - Missing nonce validation can lead to replay attacks
+
+2. `payload`:
+   - Contains the data being sent to the destination chain
+   - Must be properly encoded/decoded between chains
+   - Use matching encoding/decoding methods (e.g., `abi.encode`/`abi.decode`)
+   - Incorrect encoding can lead to undefined behavior or data corruption
+
+3. `consistencyLevel`:
+   - Determines the security level and processing delay
+   - Higher levels = better security but longer delays
+   - Lower levels = faster processing but vulnerable to block reorgs
+   - Must be set appropriately based on chain finality guarantees
+
+Reference bugs : 
+- [P1-H-01] (https://2301516674-files.gitbook.io/~/files/v0/b/gitbook-x-prod.appspot.com/o/spaces%2FNGiifMbrcug9ogAfisQY%2Fuploads%2FPxfV4xPmOCVKng8LcjiH%2FMayan_MCTP_Sec3.pdf?alt=media&token=62699afe-9e67-44fb-96fe-b593041365f4) - Example of incorrect decode on destination chain
+
+
+### Message Fee Handling
+
+The `publishMessage()` function is payable and may charge fees. This is a critical consideration when implementing cross-chain operations.
+
+#### Key Considerations
+
+1. **Fee Management**
+   - Never hardcode `msg.value` for the function call
+   - Always fetch the current fee using `messageFee()` on the Wormhole-Core contract
+   - Use the returned value as `msg.value` in the `publishMessage()` call
+   - Hardcoded values may cause transaction failures if Wormhole updates fees
+
+2. **Common Implementation Bug**
+   ```solidity
+   // Problematic Implementation
+   ICircleIntegration(wormholeCircleBridge).transferTokensWithPayload(
+       transferParameters, 
+       0,  // No fee value passed
+       abi.encode(msg.sender)
+   );
+   ```
+
+   This implementation will fail when:
+   - The destination chain has message fees enabled
+   - The `msg.value` is not sufficient to cover the required fee
+   - The transaction will revert due to insufficient funds
+
+3. **Impact**
+   - Token transfers will fail on chains with enabled message fees
+   - Users may lose gas fees on failed transactions
+   - Protocol functionality breaks on specific chains
+
+4. **Correct Implementation**
+   ```solidity
+   uint256 messageFee = wormholeCore.messageFee();
+   ICircleIntegration(wormholeCircleBridge).transferTokensWithPayload{value: messageFee}(
+       transferParameters,
+       messageFee,
+       abi.encode(msg.sender)
+   );
+   ```
```

Comment:
This whole point has too much fluff and too much text. You can explain it in 1 or 2 simple sentences where you highlight the fact that the fee should be fetched by calling `messageFee()` function and that's it. 

No need to add something as obvious as the fact that "Users may lose gas fees on failed transactions" or general points that don't say much like "Protocol functionality breaks on specific chains".